pilot/pkg/model/endpointshards.go

```diff
@@ -318,10 +318,11 @@ func (e *EndpointIndex) UpdateServiceEndpoints(
 				newIstioEndpoints = append(newIstioEndpoints, nie)
 			} else {
 				// If the endpoint does not exist in shards that means it is a
-				// new endpoint. Always send new endpoints even if they are not healthy.
+				// new endpoint. Always send new healthy endpoints.
+				// Also send new unhealthy endpoints when SendUnhealthyEndpoints is enabled.
 				// This is OK since we disable panic threshold when SendUnhealthyEndpoints is enabled.
-				// Without SendUnhealthyEndpoints we do not need this; headless services will trigger the push in the Kubernetes controller.
-				if features.SendUnhealthyEndpoints.Load() {
+				// For headless services, the Kubernetes controller will trigger the push .
```

Comment:
I cannot see how this is related to headless service

Reply:
I don't think it's related, just to preserve the comment

Reply:
Looks  out of date